### PR TITLE
keep track of all time highest score

### PR DESCRIPTION
**Pull request by brettallenyo**
https://screenshot.googleplex.com/ebZrQAukqGC

Very simply keeps track of the all time high score of all users (not a user-by-user thing)

Your score resets if you get one wrong so it is about the most in a row

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/OfficeGame.java [hunk 1]
```diff
@@ -55,7 +61,20 @@
   /** Is true or false based on whether the user got the last question right */
   private boolean lastSubmission = true;
 
-  private int score = 0;
+  private long score = 0;
+
+  private DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
+  private long highScore = -1;
```
Comment (chen-dawn) [inline portfolio/src/main/java/com/google/sps/servlets/OfficeGame.java:68]: move this `highScore` above `datastore`. Variables of the same type should stay together.

File: portfolio/src/main/java/com/google/sps/servlets/OfficeGame.java [hunk 2]
```diff
@@ -72,20 +91,30 @@ public String getAnswer() {
   public void submitAnswer(String guess) {
     lastSubmission = values.get(currentPrompt).equals(guess);
     if(lastSubmission){
-        score += 1;
+      score += 1;
```
Comment (chen-dawn) [inline portfolio/src/main/java/com/google/sps/servlets/OfficeGame.java:94]: Do you need to use the datastore for this? High score is just a single number right? Now you're storing in your store a list of contiguous integers (since high score will always increase by 1 each time). You probably only need to use the datastore when you want to store the highScore for each user (where the data will be [<userId1, highScore>, <userId2, highScore>].

File: portfolio/src/main/java/com/google/sps/servlets/OfficeGame.java [hunk 3]
```diff
@@ -72,20 +91,30 @@ public String getAnswer() {
...
+      score = 0;
     }
   }
 
   public boolean getWinStatus() {
     return lastSubmission;
   }
 
-  public int getScore() {
+  public long getScore() {
```
Comment (chen-dawn) [inline portfolio/src/main/java/com/google/sps/servlets/OfficeGame.java:110]: why did you change the type to long?

File: portfolio/src/main/java/com/google/sps/servlets/OfficeGame.java [hunk 4]
```diff
@@ -55,7 +61,20 @@
...
+  private long score = 0;
+
+  private DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
+  private long highScore = -1;
+
+  public OfficeGame(){
+    Query query = new Query("Score").addSort("score", SortDirection.DESCENDING);
+    PreparedQuery results = datastore.prepare(query);
+    for(Entity entity : results.asIterable()){
```
Comment (chen-dawn) [inline portfolio/src/main/java/com/google/sps/servlets/OfficeGame.java:73]: I think data store is not needed here (see comment below). But in the implementation here, you probably only want the first entry of your `results` since that's the all time highest score.

File: portfolio/src/main/java/com/google/sps/servlets/OfficeGame.java (hunk 2 above)
Comment (brettallenyo) [reply portfolio/src/main/java/com/google/sps/servlets/OfficeGame.java:94]: Yes it is true I just want to store a single integer, but what is the alternative to doing that other than Datastore?

File: portfolio/src/main/java/com/google/sps/servlets/OfficeGame.java (hunk 3 above)
Comment (brettallenyo) [reply portfolio/src/main/java/com/google/sps/servlets/OfficeGame.java:110]: for some reason when you insert an integer into datastore it stores it and returns it as a long

File: portfolio/src/main/java/com/google/sps/servlets/OfficeGame.java (hunk 4 above)
Comment (brettallenyo) [reply portfolio/src/main/java/com/google/sps/servlets/OfficeGame.java:73]: I changed it so that it only accesses and changes the first entry, it never creates a new one so we do not have a bunch of newly created entities everytime the high score is increased

File: portfolio/src/main/java/com/google/sps/servlets/OfficeGame.java (hunk 2 above)
Comment (chen-dawn) [reply portfolio/src/main/java/com/google/sps/servlets/OfficeGame.java:97]: You can just use the existing variable `highScore`?

File: portfolio/src/main/java/com/google/sps/servlets/OfficeGame.java [hunk 5]
```diff
@@ -62,16 +62,19 @@
...
 
   private DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
 
-  private long highScore = -1;
-
   public OfficeGame(){
-    Query query = new Query("Score").addSort("score", SortDirection.DESCENDING);
+    Query query = new Query("HighScore").addSort("high score", SortDirection.DESCENDING);
     PreparedQuery results = datastore.prepare(query);
+    System.out.println("GOT HERE!!");
```
Comment (chen-dawn) [inline portfolio/src/main/java/com/google/sps/servlets/OfficeGame.java:73]: Remove print statements.
